cpp/src/arrow/filesystem/localfs_test.cc

```diff
@@ -472,6 +472,24 @@ TYPED_TEST(TestLocalFS, StressGetFileInfoGenerator) {
   }
 }
 
+TYPED_TEST(TestLocalFS, NeedsExtendedFileInfo) {
+  // Test setting needs_extended_file_info to true and false
```

Comment:
Comment isn't accurate. I think the true case is well covered by other tests though.
```suggestion
  // Test setting needs_extended_file_info to false
```